.github/workflows/build-images.yml

```diff
@@ -66,6 +65,7 @@ jobs:
       cache-directive: ${{ steps.selective-checks.outputs.cache-directive }}
       default-branch: ${{ steps.selective-checks.outputs.default-branch }}
       default-constraints-branch: ${{ steps.selective-checks.outputs.default-constraints-branch }}
+      runs-on: ${{ steps.selective-checks.outputs.runs-on }}
```

Comment:
This should also save quite a lot of "self-hosted" time -  building images will now be done on public runners for non-commiter runs.